app/src/main/java/org/sagebionetworks/research/mpower/tracking/view_model/TrackingTaskViewModel.java

```diff
@@ -29,63 +36,154 @@
  */
 public abstract class TrackingTaskViewModel<ConfigType extends TrackingItemConfig, LogType extends TrackingItemLog>
         extends ViewModel {
-    protected LiveData<Boolean> selectionMade;
+    private static final Logger LOGGER = LoggerFactory.getLogger(TrackingTaskViewModel.class);
+
+    public static final String LOGGING_COLLECTION_IDENTIFIER = "trackedItems";
 
     // invariant availableElements.getValue() != null
+    @NonNull
     protected MutableLiveData<Map<TrackingSection, Set<TrackingItem>>> availableElements;
 
-    // invariant activeElements.getValue() != null
-    protected MutableLiveData<Set<ConfigType>> activeElements;
+    @NonNull
+    protected Map<String, TrackingItem> trackingItemsById;
+
+    @NonNull
+    // invariant activeElementsById.getValue() != null
+    protected MutableLiveData<Map<String, ConfigType>> activeElementsById;
+
+    @NonNull
+    // invariant loggedElementsById.getValue() != null
+    protected MutableLiveData<Map<String, LogType>> loggedElementsById;
 
-    // invariant unconfiguredElements.getValue() != null
-    protected LiveData<Set<ConfigType>> unconfiguredElements;
+    protected Instant startDate;
 
-    // invariant loggedElements.getValue() != null
-    protected MutableLiveData<Set<LogType>> loggedElements;
+    protected Instant endDate;
 
+    @NonNull
     protected TrackingStepView stepView;
 
-    protected TrackingTaskViewModel(@NonNull final TrackingStepView stepView) {
+    protected TrackingTaskViewModel(@NonNull final TrackingStepView stepView,
+            @Nullable final LoggingCollection<LogType> previousLoggingCollection) {
         this.stepView = stepView;
-        this.availableElements = new MutableLiveData<>();
-        this.availableElements.setValue(stepView.getSelectionItems());
-        this.activeElements = new MutableLiveData<>();
-        this.activeElements.setValue(new HashSet<>());
-        this.selectionMade = Transformations.map(this.activeElements, (elements) -> !elements.isEmpty());
-        this.unconfiguredElements = Transformations.map(this.activeElements, (elements) -> {
-            Set<ConfigType> result = new HashSet<>();
-            for (ConfigType config : elements) {
-                if (!config.isConfigured()) {
-                    result.add(config);
-                }
+        availableElements = new MutableLiveData<>();
+        activeElementsById = new MutableLiveData<>();
+        loggedElementsById = new MutableLiveData<>();
+        availableElements.setValue(stepView.getSelectionItems());
```

Comment:
It seems counter-intuitive to the ViewModel architecture to determine the start state of a ViewModel based on a View, no?  The whole point of a ViewModel is to de-couple the view and model

Reply:
The stepView isn't actually a View in the sense of it being something that inherits from android.view.View. It is actually the sanitized version of the domain layer step object. This has been causing some confusion in on-boarding people and there has been some discussion about renaming it.